### DISCUSSION TITLE
The first line of setting up a new repo breaks

### DISCUSSION THREAD

**Discussion by KennethKinLum** [community/community#29278] [Other Feature Feedback, Questions, & Ideas]
The instruction you tell other people to set up a new repo:
```bash
echo "# TryFlowJS" >> README.md
git init
git add README.md
git commit -m "first commit"
git branch -M main
git remote add origin git@github.com:nonopolarity/TryFlowJS.git
git push -u origin main
```
The first line `echo "# TryFlowJS" >> README.md` may break if the Bash is set up so that if `README.md` doesn't exist yet, it will error out and say file not exist, and there is no file in the directory and the rest of the commands cannot work well with no files in the folder.

So to fix it, you could use `echo "# TryFlowJS" > README.md` but on some Bash set up, it will overwrite the `README.md`, so that's not good either.

So you could do
```bash
touch README.md
echo "# TryFlowJS" >> README.md
```
but I suppose it can be an issue on Windows machines.  It helps people using a Mac or Linux, and it doesn't do any harm on a Windows machines though.

**Comment by airtower-luna** [comment]
> The first line echo "# TryFlowJS" >> README.md may break if the Bash is set up so that if README.md doesn't exist yet, it will error out and say file not exist

Under which circumstance did you encounter this? It seems really weird, because the [relevant Bash reference manual section](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Appending-Redirected-Output) clearly says "If the file does not exist it is created" without any condition, and I've never encountered any other behavior in well over two decades of Linux use.

**Reply by KennethKinLum** [reply]
> > The first line echo "# TryFlowJS" >> README.md may break if the Bash is set up so that if README.md doesn't exist yet, it will error out and say file not exist
> 
> Under which circumstance did you encounter this? It seems really weird, because the [relevant Bash reference manual section](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Appending-Redirected-Output) clearly says "If the file does not exist it is created" without any condition, and I've never encountered any other behavior in well over two decades of Linux use.

ok, I tried, and I seemed to be due to my 
```bash
set -o noclobber
```
in `.bashrc` or `.zshrc`
It is actually quite common to use that because it'd prevent `results.txt` to be overwritten in the case of:
```bash
./runSomething.sh > results.txt
```
It can be important as a user can spend an hour to do number crunching to get `results.txt`, and one accidental line like above will overwrite it without any possibility of getting it back.

**Reply by airtower-luna** [reply]
Thank you for pointing out `noclobber`, I learned something new! :smile_cat: 

However I can't reproduce the error you describe with `>>`:

```
$ set -o noclobber
$ echo x >> x
$ cat x 
x
$ echo x >> x
$ cat x 
x
x
$ echo x > x
bash: x: cannot overwrite existing file
```

Overwriting with `>` gets blocked, appending works regardless of whether the file exists. Which makes sense to me. I wonder what's the difference? Since you mentioned zsh, maybe you use a shell other than Bash and they behave differently there? :thinking:

**Reply by KennethKinLum** [reply]
> However I can't reproduce the error you describe with `>>`:
> 
> ```
> $ set -o noclobber
> $ echo x >> x
> $ cat x 
> x
> ```
> 
> Overwriting with `>` gets blocked, appending works regardless of whether the file exists. Which makes sense to me. I wonder what's the difference? Since you mentioned zsh, maybe you use a shell other than Bash and they behave differently there? 🤔

Are you using a Mac?  Are you sure you are doing that to a new file?
> ```
> $ set -o noclobber
> $ echo x >> new_file.txt
> $ cat x 
> x
> ```

**Reply by airtower-luna** [reply]
Nope, Bash on Linux. Yes, definitely a new file.

And that's why I'm asking, different shells might have different behavior outside POSIX requirements. Bash interpreting options different depending on OS would surprise me, but it's not like I've tried on Mac. :woman_shrugging: And Zsh is completely different.

**Reply by KennethKinLum** [reply]
> Nope, Bash on Linux. Yes, definitely a new file.
> 
> And that's why I'm asking, different shells might have different behavior outside POSIX requirements. Bash interpreting options different depending on OS would surprise me, but it's not like I've tried on Mac. 🤷‍♀️ And Zsh is completely different.

That's strange. I am trying and I can only reproduce on zsh right now.  But a couple years ago I was still using bash on the Mac as it was the default shell at the time and I remember getting the same issue but right now it doesn't reproduce on bash but only zsh.

**Reply by KennethKinLum** [reply]
ok I am trying on a CentOS Stream 8 machine, and it is the same behavior:
on Bash, it doesn't happen, but on Zsh, I did `set -o noclobber` and then `echo hello >> new_file123.txt` and then it'd say 
```bash
zsh: no such file or directory: new_file123.txt
```
I am not sure why I can't reproduce on Bash any more, as 2, 3 years ago I was still using Bash and had this same issue.

**Reply by airtower-luna** [reply]
Peculiar! Maybe the Bash devs decided it was a bug, or Apple patched the version of Bash they provided while it was the default… but I'm speculating here. :grin:
